Formula/v/vineyard.rb

```diff
@@ -37,19 +37,31 @@ class Vineyard < Formula
 
   fails_with gcc: "5"
 
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/d6/4f/b10f707e14ef7de524fe1f8988a294fb262a29c9b5b12275c7e188864aed/setuptools-69.5.1.tar.gz"
+    sha256 "6c1fccdac05a97e598fb0ae3bbed5904ccb317337a51139dcd51453611bbb987"
+  end
+
   def install
     python = "python3.12"
+    venv = virtualenv_create(libexec, python)
+    venv.pip_install resources
     # LLVM is keg-only.
     ENV.prepend_path "PYTHONPATH", Formula["llvm"].opt_prefix/Language::Python.site_packages(python)
 
     # Work around an Xcode 15 linker issue which causes linkage against LLVM's
     # libunwind due to it being present in a library search path.
     ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
 
+    # Work around API changes in `apache-arrow` 16+.
+    # Issue ref: https://github.com/v6d-io/v6d/issues/1881
+    arrow_uri_files = %w[modules/io/io/local_io_adaptor.cc modules/io/io/io_factory.cc]
+    inreplace arrow_uri_files, "arrow::internal::Uri", "arrow::util::Uri"
```

Comment:
In a follow up PR, I will switch to upstream commit https://github.com/v6d-io/v6d/commit/e8b8c828f54e16163c98a9b91068f3344608431a. The changes are the same, i.e. replacing all `arrow::internal::Uri*` usage with `arrow::util::Uri*`.